File: Latihan 2/bahasaphp/Memory.php
<?php 
 //derived class 2
class Memory extends Hardware{
	
	//atribut private
	private $frequency;
	private $size;
	private $cuda;
	
	//constructor
	function __construct(){
	}
		
	//proses get dan set
	function setFrequency($frequency){
		$this->frequency = $frequency;
	}
		
	function getFrequency(){
		return $this->frequency;
	}
		
	function setSize($size){
		$this->size = $size;
	}
		
	function getSize(){
		return $this->size;
	}
	
	function setCuda($cuda){
		$this->cuda = $cuda;
	}
		
	function getCuda(){
		return $this->cuda;
	}
	
	//destructor
	 function __destruct(){
	 }
}
?>

File: Latihan 2/bahasac++/Memory.cpp
class Memory : public Hardware{ //derived class 2
	
	//atribut
	private:
	string frequency;
	string size;
	string cuda;
	
	//constructor
	public:
	
	Memory(){
	}
	
	//proses get dan set
	void setFrequency(string frequency){
		this->frequency = frequency;
	}
	
	string getFrequency(){
		return this->frequency;
	}
	
	void setSize(string size){
		this->size = size;
	}
	
	string getSize(){
		return this->size;
	}
	
	void setCuda(string cuda){
		this->cuda = cuda;
	}
	
	string getCuda(){
		return this->cuda;
	}
	
	//destructor
	~Memory(){
	}
};
	
	